pkg/monitoring/domainstats/prometheus/prometheus.go

```diff
@@ -460,6 +460,32 @@ func (metrics *vmiMetrics) updateNetwork(netStats []stats.DomainStatsNet) {
 	}
 }
 
+func (metrics *vmiMetrics) updateFilesystem(vmFSStats k6tv1.VirtualMachineInstanceFileSystemList) {
+	fsLabels := []string{"disk"}
+
+	for _, fsStat := range vmFSStats.Items {
```

Comment:
nit: Maybe return if `len(vmFSStats.Items) == 0`? This would prevent allocating `fsLabels := []string{"disk"}` for these scenarios.